### PR TITLE
Run SQLite query planner optimization at shutdown/restart

### PR DESCRIPTION
**Pull request by ferferga**
Runs ``PRAGMA optimize`` after disposing Jellyfin's application host. According to SQLite docs, this should be run at the end of the connection, when it's most suitable to run the query planner optimizations.

Source https://www.sqlite.org/pragma.html#pragma_optimize and https://sqlite.org/lang_analyze.html

EFCore arbitrarily creates or reuses connections for database contexts (based in EF6, not EFCore documentation I could find, https://docs.microsoft.com/en-us/ef/ef6/fundamentals/connection-management) but I assume that with SQLite provider (given all the concurrency issues it has) the same connection is reused no matter what, so we're taking full advantage of the query planner. **More insights on this are appreciated**.

### CODE REVIEW

**Comment by PrplHaz4** [comment]
Any idea how long this takes to run, and if it's going to be a problem for a docker container which may get force killed if it takes too long to shutdown?

**Comment by crobibero** [comment]
I just ran `PRAGMA optimize` on my 1.1G `library.db` and my `jellyfin.db`, it was basically instant
```
sqlite3 library.db "PRAGMA optimize"  0.00s user 0.00s system 83% cpu 0.005 total
sqlite3 jellyfin.db "PRAGMA optimize"  0.00s user 0.01s system 38% cpu 0.013 total                                                                                                                      
```

**Comment by ferferga** [comment]
In my experience, optimize takes more time the longer it's been since the last run. Vacuumed databases are also faster to optimize (old databases are vacuumed after each start, ``jellyfin.db`` is in the scheduled task).

Old databases are not optimized right now with the query planner, as I think it's better to have EFCore on them. ``library.db`` is by far the largest one and its not optimized (just vacuumed as I mentioned above) so this is unlikely to cause any problem, at least for now.

File: Jellyfin.Server/Program.cs [hunk 1]
```diff
@@ -221,6 +223,14 @@ private static async Task StartApp(StartupOptions options)
             finally
             {
                 appHost.Dispose();
+                _logger.LogInformation("Running query planner optimizations in the database... This might take a while");
+
+                // Run after disposing the application
+                using var context = new JellyfinDbProvider(appHost.ServiceProvider, appPaths).CreateContext();
```
Comment (cvium) [inline Jellyfin.Server/Program.cs:229]: You're accessing a property on a disposed object

Comment (ferferga) [reply Jellyfin.Server/Program.cs:229]: @cvium Oh true. But how is possible this works then? This means we're not disposing it properly as it's still available?

I guess switching the order and running this before dispose is enough anyway.

Comment (cvium) [reply Jellyfin.Server/Program.cs:229]: Being disposed doesn't mean it's completely gone. It has to be picked up by the GC.

**Review by cvium — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Pull request merged**